### PR TITLE
puppet-in-google-cloud-platform

### PR DESCRIPTION
**Pull request by knsakib**
I have tried to write a tutorial how Puppet Master-Agent configuration can be implemented in Google Cloud Compute Engine resources. It is a very simple web server but demonstrate functionality and a road-map of a complex devops practice.       

- [ ] Tech writing / styleguide review - @ccarpentiere
- [ ] Formatting review for publishing - @jmdobry

### CODE REVIEW

**Comment by knsakib** [comment]
I signed it!

File: tutorials/puppet-in-google-cloud-platform.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+2. We'll use the --all option to sign certificate:
+        sudo /opt/puppetlabs/bin/puppet cert sign --all
+
+You can also do a single sign by
+        sudo /opt/puppetlabs/bin/puppet cert sign puppet-agent.c.YOUR_PROJECT_ID.internal
+
+## Write a simple webserver module and Manifest that will install Apache2 and write hello world page
+
+1. In the puppet-master, go to folder /etc/puppetlabs/code/environments/production/manifests/ and make a manifest file site.pp as
```
Comment (ccarpentiere) [inline tutorials/puppet-in-google-cloud-platform.md]: On the **puppet-master** instance, navigate to the folder `/etc/puppetlabs/code/environments/production/manifests/`, make a manifest file named `site.pp`, and copy the following code into it:

File: tutorials/puppet-in-google-cloud-platform.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        node /agent/{
+          include webserver
+        }
+
+2. Now go to modules directory by
```
Comment (ccarpentiere) [inline tutorials/puppet-in-google-cloud-platform.md]: 2. Navigate to the `/etc/puppetlabs/code/environments/production/modules` directory and make a new directory by running the following command:

     `sudo mkdir -p webserver/manifests`

File: tutorials/puppet-in-google-cloud-platform.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        cd /etc/puppetlabs/code/environments/production/modulesand
+
+then make a directory by
+        sudo mkdir -p webserver/manifests
+
+3. In the above manifests directory create a file init.pp as
```
Comment (ccarpentiere) [inline tutorials/puppet-in-google-cloud-platform.md]: In the new `manifests` directory, create a file named `init.pp`, and copy the following code into it:

File: tutorials/puppet-in-google-cloud-platform.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        class webserver {
+          package { 'apache2':
+          ensure => present
+        }
+        file {'/var/www/html/index.html': # resource type file and filename
+          ensure => present, # make sure it exists
+          content => "<h1>This page is installed from Puppet Master</h1>", # content of the file
+        }
+      }
+4. Run the following command in the Puppet Agent to get the Catalog from the Puppet Master to apply the manifest.
```
Comment (ccarpentiere) [inline tutorials/puppet-in-google-cloud-platform.md]: Run the following command on the **puppet-agent** instance to get the catalog from the Puppet master and apply the manifest:

File: tutorials/puppet-in-google-cloud-platform.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
```
Comment (ccarpentiere) [inline tutorials/puppet-in-google-cloud-platform.md]: Specify the command to get the catalog

File: tutorials/puppet-in-google-cloud-platform.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+5. Copy and Paste the external IP of the Puppet Agent in your browser. We should see simple web page with a message: This page is installed from Puppet Master
```
Comment (ccarpentiere) [inline tutorials/puppet-in-google-cloud-platform.md]: Copy the external IP address of the **puppet-agent** instance (you can get this from the **VM instances** page in the GCP console) and paste it into your browser. You should see a web page with the message "**This page is installed from Puppet Master**".

**Comment by jmdobry** [comment]
@ccarpentiere Does this look good to you now?

**Pull request merged**